src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -39,6 +41,14 @@ internal sealed class EnumConverter<T> : JsonPrimitiveConverter<T>
         // This is used to prevent flooding the cache due to exponential bitwise combinations of flags.
         // Since multiple threads can add to the cache, a few more values might be added.
         private const int NameCacheSizeSoftLimit = 64;
+        private const string NumericPattern = @"^\s*(\+|\-)?[0-9]+\s*$";
+
+#if NETCOREAPP
+        [GeneratedRegex(NumericPattern)]
```

Comment:
Given that the regex is applied to potentially untrusted code, we should be specifying a match timeout. Most guides dictate using a 100-200 ms upper bound.

Reply:
Fixed.